### DISCUSSION TITLE
How can I achieve a clean "text reflow" - similar to Canva ?

### DISCUSSION THREAD

**Discussion by amrithy** [fabricjs/fabric.js#9494] [Q&A]
Hi - I was looking at a feature in canva where the textbox would reflow and auto-adjust the other objects in the group. 
It seems to be using a combination of "anchoring" (which most likely maps to originX, originY in fabric.js) - and "grouping". 

I'm attaching a video to explain better - https://drive.google.com/file/d/17aEQuBX1tkdC6Sq0AN8-q-t-IZGv8lk4/view?usp=sharing

Specific aspects that I am very keen on:
* When multiple textboxes are grouped together - each textbox reflows based on the text content provided and the "anchor point". Additionally, the other textboxes position is adjusted in the direction of the reflow. 
* When there are textboxes and other objects (like a shape - rectangle for example) grouped together - the shape object width/height adjusts based on the text reflow. 

I do understand that these may not just work out-of-the-box with fabric's implementation of "obejct origin" and "groups". I would still be interested in understanding how I should approach such an implementation. 
Any pointers and suggestions on how I could achieve this behaviour will be really helpful!

**Comment by ShaMan123** [comment]
I see 2 things you need to do:

1. Have an anchor point for `onInput`:
```ts
class Textbox extends fabric.Textbox {
  onInput(e) {
    const origin = this.getRelativeXY();
    super.onInput(e);
    this.setRelativeXY(origin);
  }
}
```
3. A custom group layout: #9148
